src/components/hooks/useStarterPackEntry.native.ts

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+
+import {createStarterPackLinkFromAndroidReferrer} from 'lib/strings/starter-pack'
+import {isAndroid} from 'platform/detection'
+import {useSetCurrentStarterPack} from 'state/preferences/starter-pack'
+import {useUsedStarterPacks} from 'state/preferences/used-starter-packs'
+import SwissArmyKnife from '../../../modules/expo-bluesky-swiss-army'
+import GooglePlayReferrer from '../../../modules/expo-google-play-referrer'
+
+export function useStarterPackEntry() {
+  const [ready, setReady] = React.useState(false)
+  const setCurrentStarterPack = useSetCurrentStarterPack()
+  const usedStarterPacks = useUsedStarterPacks()
+  const hasRan = React.useRef(false)
+
+  React.useEffect(() => {
+    if (ready || hasRan.current) return
+
+    hasRan.current = true
+    ;(async () => {
+      let uri: string | null | undefined
+
+      if (isAndroid) {
+        const res = await GooglePlayReferrer.getReferrerInfoAsync()
+
+        if (res && res.installReferrer) {
+          uri = createStarterPackLinkFromAndroidReferrer(res.installReferrer)
+        }
+      } else {
+        uri = await SwissArmyKnife.getStringValueAsync('starterPackUri', true)
```

Comment:
Same question here. If anything fails we'll be stuck with a splash screen so we need high confidence, right?

Reply:
yeah +1 — 500ms timeout seems good, could pry use a try/catch and logging here too

Reply:
yup, i made it 500ms, though sould never last that long.

idt there is anything we'll catch with a try/catch here. `NSUserDefaults` will just either give us a string or null - either of which is a valid response here for us since we don't require the string to be set.